components/esp_modem/Kconfig

```diff
@@ -76,4 +76,11 @@ menu "esp-modem"
         help
             If enabled, APIs to add URC handler are available
 
+    config ESP_MODEM_ADD_DEBUG_LOGS
```

Comment:
We could just document the tags and set the `ESP_LOG_BUFFER_HEXDUMP` to verbose level. User would only need to set the tag level to verbose in their code, making this config option unnecessary.

Reply:
Make sense, but I'm actually using these logs very often when supporting users with some new/unsupported device and update the library based on shared logs.

we could make it verbose, but the important logs would be just diluted in tons of other unrelated messages,
we could ask users to add log filtering in their applications, but that would again complicate things.

this is just a simple option they could turn on and share more-less compact logs.